File: java/src/main/java/io/fivexer/sdk/model/WorkerLogin.java
package io.fivexer.sdk.model;

import io.fivexer.sdk.internal.Json;

/**
 * Worker portal credentials.
 *
 * <p>Fluent builder: set what you need, then hand it to the client. {@link #toJson()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
public class WorkerLogin {
    private String workspaceId;
    private String workerId;
    private String pin;

    public WorkerLogin(String workspaceId, String workerId, String pin) {
        if (workspaceId == null) {
            throw new IllegalArgumentException("workspaceId is required");
        }
        if (workerId == null) {
            throw new IllegalArgumentException("workerId is required");
        }
        if (pin == null) {
            throw new IllegalArgumentException("pin is required");
        }
        this.workspaceId = workspaceId;
        this.workerId = workerId;
        this.pin = pin;
    }

    public String getWorkspaceId() { return workspaceId; }
    public String getWorkerId() { return workerId; }
    public String getPin() { return pin; }



    /** Serialise for the wire, omitting unset fields. */
    public String toJson() {
        return Json.write(this);
    }
}

File: php/src/Model/WorkerLogin.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Worker portal credentials.
 *
 * Fluent builder: set what you need, then hand it to the client. {@see self::toArray()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
final class WorkerLogin
{
    public function __construct(
        private readonly string $workspaceId,
        private readonly string $workerId,
        private readonly string $pin,
    ) {
    }


    public function getWorkspaceId(): string
    {
        return $this->workspaceId;
    }

    /** The worker this login authenticates as; adopted by the client on success. */
    public function getWorkerId(): string
    {
        return $this->workerId;
    }

    public function getPin(): string
    {
        return $this->pin;
    }

    /**
     * Serialise for the wire, omitting unset fields.
     *
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'workspaceId' => $this->workspaceId,
            'workerId' => $this->workerId,
            'pin' => $this->pin,
        ] + Json::compact([

        ]);
    }
}
